### ISSUE TITLE
[$60] Replace multiple template/types/categories calls with single call to get all data

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#2472]
For getting all the templates/types/categories data we call multiple public endpoints - 

- `/v4/projects/metadata/projectTemplates`
- `/v4/projects/metadata/productTemplates`
- `/v4/projects/metadata/projectTypes`
- `/v4/projects/metadata/productCategories`

As now we have support in our backend to replace all these calls with single call 

- `GET /v4/projects/metadata`

Ref - https://github.com/topcoder-platform/tc-project-service/issues/161

**Comment by maxceem** [comment]
As we do it, probably it also makes sense to adjust the Redux Store structure. 

Currently, Redux Store looks like this:
```js
{
   projectState: {
      projectTemlate: {}, // current project's projectTemplate
      productTemplates: [],  // current project's productTemplates
      allProductTemplates: [], // all possbile productTemplates <= /v4/projects/metadata/productTemplates
   },
   templates: {
      projectTemplates: [...]   // <= /v4/projects/metadata/projectTemplates
      isProjectTemplatesLoading: false 
      projectCategories: [...]  // <= /v4/projects/metadata/projectTypes
      isProjectCategoriesLoading: false 
   }
}
```

It can be refactored to:
```js
{
   projectState: {
      // don't keep templates data anymore
   },
   templates: {
      // stay as before
      projectTemplates: [...]   // <= /v4/projects/metadata/projectTemplates
      projectCategories: [...]  // <= /v4/projects/metadata/projectTypes
      isLoading: false // instead of `isProjectTemplatesLoading` and `isProjectCategoriesLoading`
      // moved from projectState, before it was allProductTemplates
      productTemplates: [...] // <= /v4/projects/metadata/productTemplates
   }
}
```
So all templates will be loaded together on the occasions when before we loaded only some templates. 

As now we will have all tempalates the values which were previously saved in `projectState.projectTemlate` and `projectState.productTemplates` could be retrieved from the lists in `templates.*` by using their ids.

Note `/v4/projects/metadata/productCategories` we currently don't use in the connect-app.

@RishiRajSahu what do you think?

**Comment by RishiRajSahu** [comment]
@maxceem So,  there are two part for this task
1.Retrieving the templates data using single call instead of multiple calls
2.Using the templates data from store - this might be require some more changes and even changes to redux-store to use it efficiently and remove redundancy

**Comment by maxceem** [comment]
As discussed with @RishiRajSahu we will split this task into two parts. In this issue we will do only the next part:
- We have to replace all actions which currently populate `projectState.allProductTemplates`, `templates.projectTemplates` or `templates.projectCategories` with one action which makes one API call and populates them all together. I. e. we replace actions which loads the full lists of templates/categories.
If these values are already in store, this action shouldn’t be called again as all data is already here and is not changing.
- Note, that as now we have only one API call we have to replace two properites in Redux Store `templates.isProjectTemplatesLoading` and `templates.isProjectCategoriesLoading` with one `templates.isLoading`.

The second part will be handled as a separate [issue #2474](https://github.com/appirio-tech/connect-app/issues/2474). The one who accomplish this task will have priority to pick up the second part of this task.

**Comment by arun1595** [comment]
@maxceem I would like to tackle this issue.

**Comment by arun1595** [comment]
@maxceem I've fixed this issue in #2494 

Check it out when you get a chance. Thanks!
